ndnfs/daemon/driver.go

```
@@ -5,68 +5,233 @@ import (
+	}
+	if conf.Chunksize == 0 {
+		conf.Chunksize = defaultChunkSize
+	}
+	if conf.Mountpoint == "" {
+		conf.Mountpoint = defaultMountPoint
+	}
+	log.Info(DN, " config: ", conf)
 	d := NdnfsDriver{
 		Scope:			"local",
 		DefaultVolSz:	1024,
-		Client:         client,
 		Mutex:          &sync.Mutex{},
+		Endpoint:       fmt.Sprintf("http://%s:%d/", conf.Nedgerest, conf.Nedgeport),
+		Config:         &conf,
 	}
 	return d
 }
 
+func (d *NdnfsDriver) Request(method, endpoint string, data map[string]interface{}) (body []byte, err error) {
+	log.Debug("Issue request to Nexenta, endpoint: ", endpoint, " data: ", data, " method: ", method)
+	if d.Endpoint == "" {
+		err = errors.New("Unable to issue requests without specifying Endpoint")
+		log.Panic(err.Error())
+		return nil, err
+	}
+	datajson, err := json.Marshal(data)
+	if (err != nil) {
+		log.Panic(err)
+	}
+
+	tr := &http.Transport{}
+	client := &http.Client{Transport: tr}
+	url := d.Endpoint + endpoint
+	req, err := http.NewRequest(method, url, nil)
+	if len(data) != 0 {
+		req, err = http.NewRequest(method, url, strings.NewReader(string(datajson)))
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Basic " + basicAuth(d.Config.Username, d.Config.Password))
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Panic("Error while handling request ", err)
+		return nil, err
+	}
+	body, err = ioutil.ReadAll(resp.Body)
+	log.Debug("Got response, code: ", resp.StatusCode, ", body: ", string(body))
+	d.checkError(resp)
+	defer resp.Body.Close()
+	if (err != nil) {
+		log.Panic(err)
+	}
+	return body, err
+}
+
+func basicAuth(username, password string) string {
+		auth := username + ":" + password
+		return base64.StdEncoding.EncodeToString([]byte(auth))
+	}
+
+func (d *NdnfsDriver) checkError(resp *http.Response) (err error) {
+	if resp.StatusCode > 399 {
+		body, err := ioutil.ReadAll(resp.Body)
+		log.Error(resp.StatusCode, body, err)
+		return err
+	}
+	return err
```

I think it won't compile. It should fail with 'undefined: err'